src/core_codemods/tempfile_mktemp.py

```diff
@@ -19,25 +23,64 @@ class TempfileMktempTransformer(
     change_description = "Replaces `tempfile.mktemp` with `tempfile.mkstemp`."
     _module_name = "tempfile"
 
-    def leave_SimpleStatementLine(self, original_node, updated_node):
+    def __init__(
+        self,
+        context: CodemodContext,
+        results: list[Result] | None,
+        file_context: FileContext,
+        _transformer: bool = False,
+    ):
+        self.mktemp_calls: set[cst.Call] = set()
+        super().__init__(context, results, file_context, _transformer)
+
+    def visit_Call(self, node: cst.Call) -> None:
+        self.mktemp_calls.add(node) if self._is_mktemp_call(node) else None
+
+    def leave_SimpleStatementLine(
+        self,
+        original_node: cst.SimpleStatementLine,
+        updated_node: cst.SimpleStatementLine,
+    ) -> cst.SimpleStatementLine | cst.FlattenSentinel:
+        if not self.node_is_selected(original_node):
+            return updated_node
+
         match original_node:
             case cst.SimpleStatementLine(body=[bsstmt]):
-                return self.check_mktemp(original_node, bsstmt)
+                if maybe_tuple := self._is_assigned_to_mktemp(bsstmt):
+                    assign_name, call = maybe_tuple
+                    return self.report_and_change(
+                        call, assign_name, original_node.leading_lines
+                    )
+                if maybe_tuple := self._mktemp_is_sink(bsstmt):
+                    wrapper_func_name, call = maybe_tuple
+                    return self.report_and_change(
+                        call,
+                        wrapper_func_name,
+                        original_node.leading_lines,
+                        assignment=False,
+                    )
+
+        # If we get here it's because there is a mktemp call but we haven't fixed it yet.
+        for unfixed in self.mktemp_calls:
+            self.report_unfixed(unfixed, reason="Pixee does not yet support this fix.")
+        self.mktemp_calls.clear()
         return updated_node
```

Comment:
This logic feels weird here.

Say a mktemp call `mcall` is contained in the statement, `leave_Call` will be called first and `mcall` will be added to `self.mktemp_calls`. 

Afterwards, `leave_simpleStatementLine` will run. If `mcall` is fixed, it will be reported as changed, but since its removal from `self.mktemp_calls` only happens at the end, it will also be reported as unfixed.

In short, can you add a test to show that no call is being reported as fixed and unfixed at the same time?

Reply:
I checked for this. The key is the `return` statements above. If the code path hits the fixed cases we have handled, it returns, never reaching this code. Easy enough to add an assert statement to demonstrate, so will do.

Reply:
see https://github.com/pixee/codemodder-python/pull/586/files#diff-d452b243674c71654239be30328a960ed2c1e0aea06bc5b57c1ab7c8169c3cb9R43

Reply:
> I checked thought I checked this. The key is the `return` statements above. If the code path hits the fixed cases we have handled, it returns, never reaching this code. Easy enough to add an assert statement to demonstrate, so will do.

The problem is when you have two or more mktemp calls. If the first one is fixed but the second one is not. When you report the second one, the first should still be included in `self.mktemp_calls`. This results in it being reported as unfixed.

Reply:
Example please. Because what you're describing sounds like a super edge case of two calls in the same line.

Reply:
Just include `self.mktemp_calls.clear()` before the returns in `leave_SimpleStatementLine` and I'll be satisfied.